R/kiss-report.R

```diff
@@ -132,16 +143,44 @@ read.KissReport <- function(report) {
   # resultsLink should now be ready
   print("Report ready, pulling results")
   results <- as.data.frame(loadPages(report, resultsLink), stringsAsFactors = FALSE)
-  if (!is.null(report$columnNames)) {
-    names(results) <- report$columnNames
-  }
+  names(results) <- names(report)
 
-  # We don't necessarily want the times in the timezone KM returns them in.
-  # Use try_convert_time to see if any of the columns are filled with times as
-  # strings. If so then convert them to POSIX.ct classes with the timezone data
-  # from FROM_TIMEZONE. try_convert_time will then force the UTC timezone for
-  # all returned time values.
-  map_dates_results <- dplyr::mutate_each_(results, dplyr::funs(try_convert_time), names(results))
-  map_dates_results
+  # KissMetrics returns times as unix timestamps (seconds from origin in UTC)
+  # Update every column generated from a calculation with type matching *_date_*
+  # to be POSIX.ct time.
+  results[, reportCalculationClasses(report) == "timestamp"] <-
+    as.POSIXct(as.numeric(results[, reportCalculationClasses(report) == "timestamp"]),
+               origin = "1970-01-01",
+               tz = "UTC")
+
+  results
+}
+
+#' @export
+asJson.KissReport <- function(report) {
+  template <- '
+  {
+    "sort":"0",
+    "order":"asc",
+    "product_id":"{{product_id}}",
+    "query_params": {
+      "type":"group",
+      "filter":{{segment}},
+      "defaultCalculationDateRange":{{defaultCalculationDateRange}},
+      "calculations":[{{calculations}}]
+    }
+  }'
+
+  json <- template
+  json <- replacePlaceholder(json, "\\{\\{product_id\\}\\}", report$productId)
```

Comment:
Why do it this way instead of using str::str_interp for this instead?
Also - we should be using %>% to pipe these calls together